### PR TITLE
DOC: Minor fixups for 2.2.0 whatsnew

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.2.0.rst [hunk 1]
```diff
@@ -306,21 +306,19 @@ Other enhancements
...
 - :meth:`Series.ffill`, :meth:`Series.bfill`, :meth:`DataFrame.ffill`, and :meth:`DataFrame.bfill` have gained the argument ``limit_area`` (:issue:`56492`)
 - Allow passing ``read_only``, ``data_only`` and ``keep_links`` arguments to openpyxl using ``engine_kwargs`` of :func:`read_excel` (:issue:`55027`)
-- DataFrame.apply now allows the usage of numba (via ``engine="numba"``) to JIT compile the passed function, allowing for potential speedups (:issue:`54666`)
 - Implement masked algorithms for :meth:`Series.value_counts` (:issue:`54984`)
 - Implemented :meth:`Series.str.extract` for :class:`ArrowDtype` (:issue:`56268`)
-- Improved error message that appears in :meth:`DatetimeIndex.to_period` with frequencies which are not supported as period frequencies, such as "BMS" (:issue:`56243`)
-- Improved error message when constructing :class:`Period` with invalid offsets such as "QS" (:issue:`55785`)
-- The dtypes ``string[pyarrow]`` and ``string[pyarrow_numpy]`` now both utilize the ``large_string`` type from PyArrow to avoid overflow for long columns (:issue:`56259`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.2.0.rst:322]: Where did this go?

Comment (rhshadrach) [reply doc/source/whatsnew/v2.2.0.rst:322]: Thanks - restored

File: doc/source/whatsnew/v2.2.0.rst [hunk 2]
```diff
@@ -433,8 +425,6 @@ Optional libraries below the lowest tested version may still work, but are not c
 +=================+=================+=========+
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:425]: Could we just remove this table and recommend the users check the ``` :ref:`install.optional_dependencies` ``` for the new minimum versions tested?

Comment (phofl) [reply doc/source/whatsnew/v2.2.0.rst:425]: Would be hard to figure out what changed if you really care about those versions

that said, probably nobody really does, so maybe just move that table all the way to the bottom of the whatsnew?

Comment (mroeschke) [reply doc/source/whatsnew/v2.2.0.rst:425]: That's fair. I think I updated these version a while back so I'll make a follow up to just have one table with the updated verions

**Review by mroeschke — APPROVED** [review]

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @rhshadrach

**Pull request merged**